src/key.js

```diff
@@ -1030,6 +1116,70 @@ function SubKey(subKeyPacket) {
   this.revocationSignatures = [];
 }
 
+/**
+ * generate a new SubKey.
+ * @param {Integer} options.numBits (optional) number of bits for the key creation.
+ * @param  {String} options.curve (optional) elliptic curve for ECC keys
+ * @param  {Date} options.date  Override the creation date of the key and the key signatures
+ *
+ * @returns {Promise<module:key.SubKey>} return the subkey if successful.
+ * @async
+ */
+SubKey.generate = async function(options) {
```

Comment:
What's the use-case for generating a standalone subkey? Otherwise this can be merged into `addSubkey`.

Reply:
This question is simulate to what's the use-case for the Subkey Class? 

We abstract the data and functions to a class is used to reuse codes mostly.